### PR TITLE
Create own git hook document

### PR DESCRIPTION
**Pull request by legionth**
Documentation about the code style git hooks.

I don't know if this the right branch to merge, just give me feedback on this :+1:

### CODE REVIEW

**Comment by legionth** [comment]
I could also create a PR after `ci_restructuring` is merged into `ILIAS/trunk` :thinking:

**Review by ILIASci — APPROVED** [review]
Looks good to me, but like we discussed, we should think about separating the code examples into files and put them into [here](https://github.com/ILIAS-eLearning/DeveloperTools/tree/master/git_hooks) maybe.

**Comment by legionth** [comment]
Created a branch on my own fork: https://github.com/legionth/DeveloperTools/tree/code-style-hooks/git_hooks/hooks

File: docs/documentation/git-hooks.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+##  Git Hooks
+
+This document will list some useful git hooks to create a better development
```
Comment (mjansenDatabay) [inline docs/documentation/git-hooks.md]: Wir sollten Begriffe wie "GIT" einheitlich nutzen, d.h. "GIT", "git", oder "Git", nur überall gleich.

**Comment by mjansenDatabay** [comment]
Who is @ILIASci :-)?

File: docs/documentation/git-hooks.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+##  Git Hooks
+
+This document will list some useful Git hooks to create a better development
```
Comment (mjansenDatabay) [inline docs/documentation/git-hooks.md]: `This document lists some useful Git hooks to create a better development`

File: docs/documentation/git-hooks.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+### Usage
+
+Preconditions:
+* [composer](https://getcomposer.org/) must be installed
+* [Git](https://git-scm.com/) must be configured for the local project and
+  the development environment
+
+__Please be aware that these recommendations are for a development environment
+only, and may not be used in actual environments__
```
Comment (mjansenDatabay) [inline docs/documentation/git-hooks.md]: `actual environments` = `in production`

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Only some minor remarks.

I also think the separation of the examples into separate files would be great.

**Comment by legionth** [comment]
> I also think the separation of the examples into separate files would be great.

Then we wouldn't need to add these hooks to the https://github.com/ILIAS-eLearning/DeveloperTools.
IMO either we put the git hook files into the main repository or the developer tools not both.

Honestly I don't know if this README belongs in the main repository either.

**Comment by gvollbach** [comment]
> Honestly I don't know if this README belongs in the main repository either.

To be honest I would put this README into the developer tools repo also and maybe put only a link, to the developer tools, at most. 

Also I would remove the source code from the README and link to the files in the delveloper tools directory.

**Comment by gvollbach** [comment]
> Who is @ILIASci :-)?

It is a really early stage of a seriously dumb AI.  ;)

**Review by mjansenDatabay — APPROVED** [review]

**Comment by legionth** [comment]
Review plx.

**Review by gvollbach — APPROVED** [review]
:+1:

**Comment by gvollbach** [comment]
Thanks!

**Pull request merged**